porch/pkg/git/git.go

```diff
@@ -676,21 +686,84 @@ func (r *gitRepository) fetchRemoteRepository(ctx context.Context) error {
 }
 
 // Verifies repository. Repository must be fetched already.
-func (r *gitRepository) verifyRepository(opts *GitRepositoryOptions) error {
+func (r *gitRepository) verifyRepository(ctx context.Context, opts *GitRepositoryOptions) error {
 	// When opening a temporary repository, such as for cloning a package
 	// from unregistered upstream, we won't be pushing into the remote so
 	// we don't need to verify presence of the main branch.
-	if !opts.SkipMainBranchVerification {
-		// Check existence of main branch. If it doesn't exist, fail.
-		// If the main branch doesn't exist, we could create draft or proposal branch
-		// which could become the default branch and those are hard to delete.
-		if _, err := r.repo.Reference(r.branch.RefInLocal(), false); err != nil {
+	if opts.MainBranchStrategy == SkipVerification {
+		return nil
+	}
+
+	if _, err := r.repo.Reference(r.branch.RefInLocal(), false); err != nil {
+		switch opts.MainBranchStrategy {
```

Comment:
I see that this can be set through the code above in cache.go or the clone package mutation. Is it ever possible that this is not set? Should this switch statement have a default case?

Reply:
It is always set, but might be the default value. The default value for the enum is `ErrorIfMissing`, just like it used to be.
All the values are covered in the code, but I've added an error message for the default case in the switch statement in case additional values are added at some point in the future.